scalafmt-core/shared/src/main/scala/org/scalafmt/rewrite/RemoveScala3OptionalBraces.scala

```diff
@@ -0,0 +1,137 @@
+package org.scalafmt.rewrite
+
+import scala.meta.Defn
+import scala.meta.Template
+import scala.meta.Term
+import scala.meta.tokens.Token
+
+import org.scalafmt.config.RewriteScala3Settings
+import org.scalafmt.config.ScalafmtConfig
+import org.scalafmt.internal.FormatToken
+import org.scalafmt.internal.FormatTokens
+import org.scalafmt.util.TreeOps
+
+object RemoveScala3OptionalBraces extends FormatTokensRewrite.RuleFactory {
+
+  override def enabled(implicit style: ScalafmtConfig): Boolean =
+    style.runner.dialect.allowSignificantIndentation &&
+      style.rewrite.scala3.removeOptionalBraces
+        .ne(RewriteScala3Settings.RemoveOptionalBraces.never)
+
+  override def create(ftoks: FormatTokens): FormatTokensRewrite.Rule =
+    new RemoveScala3OptionalBraces(ftoks)
+
+}
+
+private class RemoveScala3OptionalBraces(ftoks: FormatTokens)
+    extends FormatTokensRewrite.Rule {
+
+  import FormatTokensRewrite._
+
+  private def allowOldSyntax(implicit style: ScalafmtConfig): Boolean =
+    ConvertToNewScala3Syntax.enabled ||
+      style.rewrite.scala3.removeOptionalBraces
+        .ne(RewriteScala3Settings.RemoveOptionalBraces.newSyntaxOnly)
+
+  override def enabled(implicit style: ScalafmtConfig): Boolean =
+    RemoveScala3OptionalBraces.enabled
+
+  override def onToken(implicit
+      ft: FormatToken,
+      style: ScalafmtConfig
+  ): Option[Replacement] = Option {
+    ft.right match {
+      case x: Token.LeftBrace // skip empty brace pairs
+          if !ftoks.nextNonComment(ftoks.next(ft)).right.is[Token.RightBrace] =>
+        ft.meta.rightOwner match {
+          case t: Term.Block if t.stats.nonEmpty => onLeftForBlock(t)
+          case t: Template if t.stats.nonEmpty || t.self.tokens.nonEmpty =>
+            if (t.parent.exists(_.is[Defn.Given])) removeToken
+            else
+              replaceToken(":")(new Token.Colon(x.input, x.dialect, x.start))
+          case _: Term.For if allowOldSyntax || {
+                val rbFt = ftoks(ftoks.matching(ft.right))
+                ftoks.nextNonComment(rbFt).right.is[Token.KwDo]
+              } =>
+            removeToken
+          case _: Term.ForYield => removeToken
+          case _: Term.Match => removeToken
+          case _: Term.Try => removeToken
+          case _: meta.Ctor.Secondary =>
+            if (!ftoks.prevNonComment(ft).left.is[Token.Equals])
```

Comment:
Procedure syntax is no longer allowed an will be disabled in the [Scala 3 dialect](https://github.com/scalameta/scalameta/pull/2313). This also doesn't compile in Scala 3.

The alternative approach would be to have the rewrite for Scala 2 dialect, but we do have it already in scalafix, which is used for example by the migration tool.